tests/unit/components/detail-test.js

```diff
@@ -70,5 +76,163 @@ describeComponent(
 
       expect(state.value).to.eql(expectedValue)
     })
+
+    describe('.storeUpdated()', function () {
+      let changeHandler, validationHandler, reduxStore, newProps
+
+      beforeEach(function () {
+        reduxStore = {
+          getState: sandbox.stub()
+        }
+        changeHandler = sandbox.stub()
+        validationHandler = sandbox.stub()
+
+        component.setProperties({
+          errors: [],
+          onChange: changeHandler,
+          onValidation: validationHandler,
+          reduxModel: {
+            fizz: 'bang'
+          },
+          reduxStore,
+          renderValue: {
+            foo: 'bar'
+          }
+        })
+      })
+
+      describe('when renderValue has changed but errors has not', function () {
+        let newValue
+        beforeEach(function () {
+          newValue = {
+            foo: 'baz'
+          }
+
+          reduxStore.getState.returns({
+            errors: [],
+            validationResult: {
+              errors: []
+            },
+            model: {
+              fizz: 'bang'
+            },
+            value: newValue
+          })
+
+          sandbox.stub(component, 'setProperties')
+          component.storeUpdated()
+          newProps = component.setProperties.lastCall.args[0]
+        })
+
+        it('should update renderValue in properties', function () {
+          expect(newProps.renderValue).to.be.eql(newValue)
+        })
+
+        it('should not update errors in properties', function () {
+          expect(newProps.errors).not.to.be.defined
+        })
+
+        it('should fire onChange', function () {
+          expect(changeHandler.lastCall.args).to.be.eql([newValue])
+        })
+
+        it('should fire onValidation', function () {
+          expect(validationHandler.lastCall.args).to.be.eql([{errors: []}])
```

Comment:
And here's the only actual test for the new functionality. The rest were just tests to ensure existing functionality.